b12x/comm/pcie/pcie_allreduce.py

```
@@ -98,7 +148,43 @@ def from_exchange_group(
+            return None
+        if _algorithm_override() == "hierarchical":
+            return None
+        capacity = max(int(max_size), ISLAND_RS_MAX_BYTES)
+        elements = capacity // torch.bfloat16.itemsize
+        try:
+            return PCIeIslandRSAllReduce(
+                exchange_group=exchange_group,
+                device=device,
+                max_elements=elements - (elements % 2),
+            )
+        except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)